apicast/src/oauth/keycloak.lua

```diff
@@ -248,7 +262,8 @@ function _M:get_token(service, client)
   -- call Keycloak authorize
   local url = self.config.token_url
 
-  local res = http_client.post(url, req_body)
+  local headers = ngx.req.get_headers()
+  local res = http_client.post(url, ngx.req.get_post_args(), { headers = headers })
```

Comment:
I don't see any test covering why it should pass all request headers to the keycloak endpoint. Why it should do that? Shouldn't it be just some whitelist of headers?